spec/controllers/V1/tasks_controller_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+describe V1::TasksController, :type => :controller do
+
+  let!(:project) { create(:project) }
+
+  describe "PUT update_task_state" do
+    let!(:task) { create(:task, project_id: project.id) }
+    let!(:update_params) do { project_id: project.id, task_id: task.id, state: "in-progress" } end
```

Comment:
Since you are referencing your `update_params` in your `it` blocks below, you don't need the `let!` on `project`, `task`, or `update_params` since they will all lazily instantiate as expected. The `!` version should be reserved for when you need to force it to instantiate.